validity/src/proof_requester.rs

```diff
@@ -371,25 +381,19 @@ impl<H: OPSuccinctHost> OPSuccinctProofRequester<H> {
         let stdin = match request.req_type {
             RequestType::Range => self.range_proof_witnessgen(request).await?,
             RequestType::Aggregation => {
-                // Fetch consecutive range proofs from the database.
-                let range_proofs_raw = self
-                    .db_client
-                    .get_consecutive_complete_range_proofs(
-                        request.start_block,
-                        request.end_block,
-                        &self.program_config.commitments,
-                        request.l1_chain_id,
-                        request.l2_chain_id,
-                    )
-                    .await?;
-                let range_proofs: Vec<SP1ProofWithPublicValues> = range_proofs_raw
-                    .iter()
-                    .map(|proof| {
-                        bincode::deserialize(proof.proof.as_ref().unwrap())
-                            .expect("Deserialization failure for range proof")
-                    })
-                    .collect();
-                self.agg_proof_witnessgen(request, range_proofs).await?
+                self.agg_proof_witnessgen(
+                    request.start_block,
+                    request.end_block,
+                    B256::from_slice(request.checkpointed_l1_block_hash.as_ref().ok_or_else(
+                        || anyhow::anyhow!("Aggregation proof has no checkpointed block."),
+                    )?),
+                    request.l1_chain_id,
+                    request.l2_chain_id,
+                    Address::from_slice(request.prover_address.as_ref().ok_or_else(|| {
+                        anyhow::anyhow!("Prover address must be set for aggregation proofs.")
+                    })?),
+                )
```

Comment:
Can't we just input request here?

Reply:
The purpose of this PR is to expose a method for generating the witness that does not require a `Request` object, but rather the constituent parts.

Otherwise, the interface/abstraction is not super clear, as agg proof witnessgen is not tied to a `Request` object.